### PR TITLE
chore: relocate gapic-generator-java pom.xml to subfolder

### PR DESCRIPTION
**Pull request by burkedavison**
@blakeli0 : This is the minimum number of edits needed to make the project work with a relocated `pom.xml`.

Let's discuss whether you'd prefer to have this stay in its own branch for now.

### CODE REVIEW

**Comment by blakeli0** [comment]
> @blakeli0 : This is the minimum number of edits needed to make the project work with a relocated `pom.xml`.
> 
> Let's discuss whether you'd prefer to have this stay in its own branch for now.

The changes look good to me. The bazel build is failing though, if all the checks are passing, we could merge it in.

File: src/test/java/com/google/api/generator/engine/JavaCodeGeneratorTest.java [hunk 1]
```diff
@@ -63,8 +63,9 @@
 import org.junit.Test;
 
 public class JavaCodeGeneratorTest {
+
   private static final String GOLDENFILES_DIRECTORY =
-      "src/test/java/com/google/api/generator/engine/goldens/";
+      "../src/test/java/com/google/api/generator/engine/goldens/";
```
Comment (meltsufin) [inline src/test/java/com/google/api/generator/engine/JavaCodeGeneratorTest.java:68]: Would it be possible to change the working directory for surefire or failsafe instead?

Comment (burkedavison) [reply src/test/java/com/google/api/generator/engine/JavaCodeGeneratorTest.java:68]: Good idea. I'll give it a try.

Comment (burkedavison) [reply src/test/java/com/google/api/generator/engine/JavaCodeGeneratorTest.java]: This suggestion gets bonus points for both simplifying the PR *and* fixing the bazel build.

**Review by meltsufin — APPROVED** [review]
Glad the working directory worked!

**Review by blakeli0 — APPROVED** [review]

**Pull request merged**